src/models/event_query.cr

```diff
@@ -7,10 +7,17 @@ class Office365::EventQuery
   end
 
   def self.from_json(data)
-    parsed_data = JSON.parse(data).as_h["value"].as_a
-    result = parsed_data.map do |event|
-      Event.from_json(event.to_json)
+    parsed_data = JSON.parse(data).as_h
+    if parsed_data.has_key?("value")
+      array_data = parsed_data["value"].as_a
+      result = array_data.map do |event|
+        Event.from_json(event.to_json)
+      end
+      new(value: result)
+    else
+      parsed_error_message = parsed_data["error"]["message"]?
```

Comment:
This will still result in an explosion if `error` doesn't exist. Try spitting this out into the above if so that you're dealing with either a:
1. response with a `value` (containing and array of `Event`s)
2. error response
3. something else